lib/Echidna/Campaign.hs

```diff
@@ -110,19 +118,31 @@ runSymWorker
   -> Maybe Text -- ^ Specified contract name
   -> m (WorkerStopReason, WorkerState)
 runSymWorker callback vm dict workerId initialCorpus name = do
+  shuffleCorpus <- shuffleIO initialCorpus
   cfg <- asks (.cfg)
   let nworkers = getNFuzzWorkers cfg.campaignConf -- getNFuzzWorkers, NOT getNWorkers
   eventQueue <- asks (.eventQueue)
   chan <- liftIO $ dupChan eventQueue
 
-  flip runStateT initialState $
-    flip evalRandT (mkStdGen effectiveSeed) $ do -- unused but needed for callseq
-      lift callback
-      void $ replayCorpus vm initialCorpus
-      symexecTxs []
-      mapM_ (symexecTxs . snd) initialCorpus
-      listenerLoop listenerFunc chan nworkers
-      pure SymbolicDone
+  if (cfg.campaignConf.workers == Just 0) && (cfg.campaignConf.seqLen == 1) then do 
+    liftIO $ putStrLn "Single-transaction symbolic verification mode started:"
+    flip runStateT initialState $
+      flip evalRandT (mkStdGen effectiveSeed) $ do -- unused but needed for callseq     
+        verifyMethods -- No arguments, everything is in this environment
+        pure SymbolicVerificationDone
+  else
+    flip runStateT initialState $
+      flip evalRandT (mkStdGen effectiveSeed) $ do -- unused but needed for callseq
+        lift callback
+        listenerLoop listenerFunc chan nworkers
+        void $ replayCorpus vm initialCorpus
+        if null shuffleCorpus then 
+          replicateM_ 10 $ symexecTxs True [] -- TODO: determine how many times to symexec here
```

Comment:
does this TODO need to be addressed?

Reply:
Not right now. I need to think about it. In any case, when running in exploration mode, the concrete workers will immediately fill the corpus which will trigger more work for the symbolic worker, so there is no big deal.